joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForFunctionsCreator.scala

```diff
@@ -219,4 +193,42 @@ trait AstForFunctionsCreator(implicit withSchemaValidation: ValidationMode) { th
         astForUnknown(node)
   }
 
+  private def astForParameters(parameters: List[RubyNode], plusOne: Boolean = false): List[Ast] = {
+    parameters.zipWithIndex.map { case (parameterNode, index) =>
+      astForParameter(parameterNode, if (plusOne) index + 1 else index)
+    }
+  }
+
+  private def statementListForOptionalParams(params: List[RubyNode]): StatementList = {
+    StatementList(
+      params
+        .collect { case x: OptionalParameter =>
+          x
+        }
+        .map(statementForOptionalParam)
+    )(TextSpan(None, None, None, None, ""))
+  }
+
+  private def astForMethodBody(body: RubyNode, optionalStatementList: StatementList): Ast = {
+    if (this.parseLevel == AstParseLevel.SIGNATURES) {
+      Ast()
+    } else {
+      body match
+        case stmtList: StatementList =>
+          astForStatementListReturningLastExpression(
```

Comment:
https://github.com/joernio/joern/pull/4241 made some changes in this behaviour, is this code duplicated? May need to reflect the similar level of implicit return behaviour in editted in #4241

Reply:
Ignore :) my changes are in `astsForImplicitReturnStatement`